core/src/main/java/org/apache/calcite/plan/volcano/TopDownRuleDriver.java

```diff
@@ -335,6 +336,14 @@ default boolean onProduce(RelNode node) {
           physicals.add(rel);
         }
       }
+
+      // No need to apply O_INPUTS if the group has not been implemented yet.
```

Comment:
What is `O_INPUTS`?

Reply:
It means OptimzeInput, which is a term in Cascades planner. I borrow it from the comment[1].

[1] https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/plan/volcano/TopDownRuleDriver.java#L338

Reply:
I see no reasons to invent obscure words when the thing can be explained in plain English.

Reply:
I would change it as you suggest. But the term O_INPUT is not invented currently. It comes from a classic paper[1] which was published in 1998.

![image](https://user-images.githubusercontent.com/37774589/97863684-57062680-1d42-11eb-8b0f-37bc5918911f.png)


[1] https://15721.courses.cs.cmu.edu/spring2019/papers/22-optimizer1/xu-columbia-thesis1998.pdf